### PR TITLE
Abort sonarwhal execution if a resource dependency is missing

### LINKED ISSUE #1014: Trying to load a wrong connector throws an error
Currently, specifying the wrong formatter prompts ask the user to install/update a package: 

Config 

image

Result
image

With a connector we get also an exception:

Config

image

Result

image

IMO `connector`s should do the same as `formatter`s and in any case the behavior should be normalized across resources.

### PR DESCRIPTION
**Pull request by sarvaje**
Fix #1014

## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/getting-started/pull-requests/#commit-messagess)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [ ] Added/Updated related tests.

### CODE REVIEW

**Comment by molant** [comment]
@sarvaje there are some errors running the tests with `resource-loader`. Can you please take a look?

**Comment by sarvaje** [comment]
sure thing!

File: packages/sonarwhal/src/lib/types/resourceerror.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { ResourceErrorStatus } from '../enums/errorstatus';
+
+export class ResourceError extends Error {
+    public status: ResourceErrorStatus;
+
+    public constructor(error: Error, status: ResourceErrorStatus, message?: string) {
+        super(error.message);
```
Comment (molant) [inline packages/sonarwhal/src/lib/types/resourceerror.ts]: This signature looks a bit weird, especially when you do later a:

```js
throw new ResourceError(new Error(`Resource ${name} isn't compatible with current sonarwhal version`), ResourceErrorStatus.NotCompatible);
```

We shouldn't have to create a new error outside the constructor. It looks like there are a few things going on with TS and inheriting from built-ins in TS:

* https://www.metachris.com/2017/01/custom-errors-in-typescript-2.1/
* https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html#support-for-newtarget


But those seem to be from TS 2.1 and 2.2, not sure a bout the current status for us in node. In any case, I think we should have something like:

```ts
public constructor(error: Error | string, status: ResourceErrorStatus) { }
```

or even 

```ts
public constructor(error: string, status: ResourceErrorStatus) { }
```

to simplify things.

**Comment by sarvaje** [comment]
@molant done!

**Pull request merged**